sync-todo/v2/client/cpp/controllers/home_controller.cpp

```diff
@@ -0,0 +1,193 @@
+
+  if (_homeControllerState.subscriptionSelection == SubscriptionSelection::allItems) {
+    _homeControllerState.subscriptionSelectionLabel = showMineButtonLabel;
+  } else if (_homeControllerState.subscriptionSelection == SubscriptionSelection::myItems) {
+    _homeControllerState.subscriptionSelectionLabel = showAllButtonLabel;
+  }
+
+  auto toggleSubscriptionsButton = ftxui::Button(&_homeControllerState.subscriptionSelectionLabel,
+                                                 [=]{ _dbManager.toggleSubscriptions();
+  });
+  toggleSubscriptionsButton = VWrap("Subscription", toggleSubscriptionsButton);
+
+  auto filters = ftxui::Checkbox("Hide completed", &_homeControllerState.hideCompletedTasks);
+  filters = VWrap("Filters", filters);
+
+  auto logoutButton = ftxui::Button("Logout",
+                                    [&]{ _appState->authManager->logOut(); });
+  logoutButton = VWrap("Auth", logoutButton);
+
+  auto quitButton = ftxui::Button("Quit", appState->screen->ExitLoopClosure());
+  quitButton = VWrap("Exit", quitButton);
+
+  auto optionsLayout = ftxui::Container::Horizontal(
+      {toggleOfflineModeButton, toggleSubscriptionsButton, filters, logoutButton, quitButton});
+
+  auto homeControllerButtonView = Renderer(optionsLayout, [=] {
+    return vbox(
+        hbox(toggleOfflineModeButton->Render(),
+             ftxui::separator(),
+             toggleSubscriptionsButton->Render(),
+             ftxui::separator(),
+             filters->Render(), ftxui::separator(),
+             logoutButton->Render(),
+             ftxui::separator(),
+             quitButton->Render()
+             ) | ftxui::border | ftxui::center | size(ftxui::WIDTH, ftxui::GREATER_THAN, 100)) ;
+  });
+
+  // Accept user inputs and create new items in the database.
+  auto inputNewTaskSummary =
+      ftxui::Input(&_homeControllerState.newTaskSummary, "Enter new task summary");
+  auto newTaskCompletionStatus = ftxui::Checkbox("Complete", &_homeControllerState.newTaskIsComplete);
+
+  auto saveButton = ftxui::Button("Save", [=] {
+    _dbManager.addNew(_homeControllerState.newTaskIsComplete, _homeControllerState.newTaskSummary);
+    _homeControllerState.newTaskSummary = "";
+    _homeControllerState.newTaskIsComplete = false;
+  });
+
+  auto newTaskLayout = ftxui::Container::Horizontal(
+      {inputNewTaskSummary, newTaskCompletionStatus, saveButton});
+
+  // Lay out and render the scrollable task list.
+  auto renderTasks = ftxui::Renderer([=] {
+    auto itemList = _homeControllerState.hideCompletedTasks ? _dbManager.getIncompleteItemList() : _dbManager.getItemList();
+    ftxui::Elements tasks;
+    // If the user has toggled the checkbox to hide completed tasks, show only the incomplete task list.
+    // Otherwise, show all items.
+    for (auto &item: itemList) {
+      std::string completionString = (item.isComplete) ? " Complete " : " Incomplete ";
+      std::string mineOrNot = (item.owner_id == userId) ? " Mine " : " Them ";
```

Comment:
It would match the declension but not the width 😅  I initially had "Theirs" but the way it lined up with "Mine" made it - unpleasant to visually parse. I guess I could try more styling options to see if I can get something satisfactory with "Theirs."

Reply:
Tweaked this a bit so we're now using "Theirs" and just added some spacing to pad in front of "Mine" so the status (Complete/Incomplete) lines up nicely.